src/Client/RetryDecider.php

```diff
@@ -5,36 +5,57 @@
 namespace Keboola\HttpExtractor\Client;
 
 use DateTimeImmutable;
-use GuzzleHttp\Exception\ConnectException;
 use GuzzleHttp\Exception\RequestException;
 use GuzzleHttp\Psr7\Request;
 use GuzzleHttp\Psr7\Response;
+use Psr\Log\LoggerInterface;
 use function in_array;
 use function Keboola\Utils\isValidDateTimeString;
 
 class RetryDecider
 {
     private const MAX_RETRIES = 5;
 
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
     public function __invoke(
         int $retries,
         ?Request $request,
         ?Response $response = null,
         ?RequestException $exception = null
     ): bool {
         if ($retries >= self::MAX_RETRIES) {
+            $this->logger->info('Max retries exceeded');
             return false;
         }
 
         if ($this->shouldAbortBasedOnRetryAfterHeader($response)) {
+            $this->logger->info('Aborting due to Retry-After header value');
             return false;
         }
 
         if ($this->isRecoverableHttpStatus($response)) {
+            if ($response) {
+                $this->logger->info(sprintf(
+                    'Retrying based on "%s" HTTP status',
+                    $response->getStatusCode()
+                ));
+            }
             return true;
         }
 
         if ($this->isRecoverableException($exception)) {
+            assert($exception instanceof RequestException);
```

Comment:
assert v kodu? Navic pokud RetryDecider vyhodi vyjimku, tak ten stacktrace nikdo nerozklicuje

Reply:
To je kvuli phpstanu, někde jsem to nedávno viděl, že se to používá místo ifu a přišlo mi to cool. 

Neměl by nikdy failnout. Ta `isRecoverableException` má typehint na `?RequestException` a propadne tam jen pokud je true (tj. ne-null).

Reply:
Proč by ho nikdo nerozklíčoval? 

```
[2018-12-28 12:00:48] CRITICAL: ErrorException:assert(): assert($uri === false) failed {"errFile":"/code/src/HttpExtractorComponent.php","errLine":41,"errCode":0,"errTrace":"#0 [internal function]: Keboola\\Component\\BaseComponent::Keboola\\Component\\{closure}(2, 'assert(): asser...', '/code/src/HttpE...', 41, Array)\n#1 /code/src/HttpExtractorComponent.php(41): assert(false, 'assert($uri ===...')\n#2 /code/src/run.php(15): Keboola\\HttpExtractor\\HttpExtractorComponent->run()\n#3 {main}","errPrevious":""} []
```

Reply:
protoze ho to vyhodi behem handlovani vyjimky a ta v tom nebude zalogovana